pdata/README.md

```diff
@@ -0,0 +1,144 @@
+# Pipeline data (pdata)
+
+Pipeline data (pdata) implements data structures that represent telemetry data in-memory. All data received 
+is converted into this format, travels through the pipeline in this format, and is converted from this format by 
+exporters when sending.
+
+Current implementation primarily uses OTLP protobuf structs as the underlying data structures for many of the 
+declared structs. We keep a pointer to OTLP protobuf in the "orig" member field. This allows efficient translation 
+to/from OTLP wire protocol. The underlying data structure is kept private so that we are free to make changes to it 
+in the future.
+
+API provided by pdata ensures that no underlying data can be used in more than one place. Each pdata instance cannot 
+contain a reference to an object that is already used somewhere else.
+
+## API naming convention
+
+### protobuf message representation in pdata
```

Comment:
One more exception is that we have "Resource" in the common package instead of "resource" package.

Reply:
We don't specify package names here. I added a small paragraph for that